### DISCUSSION TITLE
What macros / variables are supported in the Open WebUI prompt editors?

### DISCUSSION THREAD

**Discussion by clang88** [open-webui/open-webui#3795] [Q&A]
In some instances it is possible to see that some prompts contain "macros" or "variables" like:

{{CONTEXT}} or {{CURRENT_USER}} and {{DATE}}. Sometimes the macros are listed like this (e.g. in the RAG prompt template): <RAGcontext>
[context] [query]

Where can I find a comprehensive list of all supported macros? I tried to find it in the code, but was unsuccessful.

**Comment by kzifla** [comment]
Also ran into the same issue. Commenting for more reach.

**Comment by clang88** [comment]
It seems like a pretty major detail, too. I tried scowering the code for any of macros using the "Mustache" notation ({{}}) but all I could find was the simple default rag template function for the [context] and [query] macros:
https://github.com/open-webui/open-webui/blob/82079e644a291349926008adffe8fe2fc3b33566/backend/apps/rag/utils.py#L191

**Comment by Mavyre** [comment]
I'm joining you on this. I'd like to make custom system prompts with more context. Couldn't find variables in the code...
